packages/malloy/src/model/malloy_query.ts

```diff
@@ -1359,16 +1376,27 @@ class QueryField extends QueryNode {
         return `(${expr.e.sql})`;
       case 'not':
         // Malloy not operator always returns a boolean
-        return `COALESCE(NOT ${expr.e.sql},TRUE)`;
+        // TODO (vitor): Sorry! I feel like woody the woodpecker saying i did not not not not not eat all the pizza
+        /**
+                In databases deep and wide,
+                Where logic twists and bits collide,
+                There lurk the double negatives,
+                Entwined in tangled predicates.
+         */
+        // I guess i can override and call super for the other cases? And put this poetic following statement in the tsql dialect...
+        return `NOT COALESCE(CASE WHEN (${expr.e.sql}) THEN 1 END, 0) = 1`;
       case 'unary-':
         return `-${expr.e.sql}`;
       case 'is-null':
         return `${expr.e.sql} IS NULL`;
       case 'is-not-null':
         return `${expr.e.sql} IS NOT NULL`;
       case 'true':
+        // TODO (vitor): check with the malloy team
+        return this.parent.dialect.booleanAsNumbers ? '1=1' : expr.node;
```

Comment:
so in other "booleanAsNumber" dialects, there are symbols "true" and "false" which evaluate to "1" and "0".

if that isn't the case for tsql then probably there should be a dialect function which returns the true and false values

```TypeScript
case 'true':
  return this.parent.dialect.booleanValue(true);
case 'false':
  return this.parent.dialect.booleanValue(false);
```

And everything except tsql uses

```
   booleanValue(v: boolean) { return v ? 'true' : 'false' }
```